### PR TITLE
refactor: remove account dependencies

### PR DESCRIPTION
**Pull request by highskore**
solves #158 to make #160 work

### CODE REVIEW

**Comment by highskore** [comment]
@kopy-kat a few issues, 

- Registry dependency is `^0.8.24`
- Seems like memory allocation for < 0.8.26 is not the best so SIMULATE=true doens't work on CI (works locally tho)

**Comment by kopy-kat** [comment]
Should we remove registry dep too?

**Comment by highskore** [comment]
> Should we remove registry dep too?

the 7702 compatibility change added to 7579 ref requries ^0.8.24 for tstore too, and 4337 is ^0.8.24

maybe lets just keep it >=0.8.24 <0.9.0 ?

**Comment by kopy-kat** [comment]
Yeah that’s mid, maybe we remove the 7579 impl too - but yeah maybe we need to stay higher than 0.8.24

**Comment by highskore** [comment]
> Yeah that’s mid, maybe we remove the 7579 impl too - but yeah maybe we need to stay higher than 0.8.24

7579 ref is removed, bumped solc to 0.8.24 >=

File: src/Accounts.sol [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.8.0 <0.9.0;
+pragma solidity >=0.8.24 <0.9.0;
```
Comment (kopy-kat) [inline src/Accounts.sol:2]: why do we need this to be higher than 24 again?

**Review by kopy-kat — APPROVED** [review]

**Comment by kopy-kat** [comment]
I'm kind of having doubts over essentially scrapping dependencies and needing to manually port over changes. These should be rare tho so it might be fine - wdyt @highskore

**Comment by highskore** [comment]
> I'm kind of having doubts over essentially scrapping dependencies and needing to manually port over changes. These should be rare tho so it might be fine - wdyt @highskore

I think it's fine since it's just accounts which should in theory only update rarely and the porting is not that bad (just updating bytecode most of the time)

File: src/Accounts.sol (hunk 1 above)
Comment (kopy-kat) [reply src/Accounts.sol:2]: should we just use sstore here in modulekit?

Comment (highskore) [reply src/Accounts.sol:2]: yea that's a good idea

**Comment by highskore** [comment]
Let's merge #162 first, will probably need to resolve some conflicts

**Comment by highskore** [comment]
@kopy-kat Found a nested dependency to erc7579 inside module-bases, do we remove the dependency? We would need to duplicate some code, it's stopping modulekit for being able to use solc 0.8.23 because of the tstore inside latest erc7579 ref

**Review by kopy-kat — APPROVED** [review]
I think this looks good to me - I think we should release it as a new major version

before merging could you try using this branch in core-modules and seeing if anything breaks?

**Pull request merged**